### PR TITLE
8328819: Remove applet usage from JFileChooser tests bug6698013

### PR DESCRIPTION
**Pull request by alisenchung**
Removing applet usage from manual JFileChooser tests
The original test is failing (nothing shows up in the JFileChooser), but this change does replace the applet usage with PassFailJFrame

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8328819](https://bugs.openjdk.org/browse/JDK-8328819): Remove applet usage from JFileChooser tests bug6698013 (**Bug** - P4)

### Reviewers
 * [Harshitha Onkar](https://openjdk.org/census#honkar) (@honkar-jdk - **Reviewer**)
 * [Tejesh R](https://openjdk.org/census#tr) (@TejeshR13 - Committer)
 * [Damon Nguyen](https://openjdk.org/census#dnguyen) (@DamonGuy - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18456/head:pull/18456` \
`$ git checkout pull/18456`

Update a local copy of the PR: \
`$ git checkout pull/18456` \
`$ git pull https://git.openjdk.org/jdk.git pull/18456/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18456`

View PR using the GUI difftool: \
`$ git pr show -t 18456`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18456.diff">https://git.openjdk.org/jdk/pull/18456.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18456#issuecomment-2015374173)

### CODE REVIEW

**Review by honkar-jdk — COMMENTED** [review]
> The original test is failing (nothing shows up in the JFileChooser), but this change does replace the applet usage with PassFailJFrame

@alisenchung The original test works on Windows. Is it on macOS that nothing is shown ?

**Review by honkar-jdk — COMMENTED** [review]
Ideally the Instruction window and Test window appear side-by-side. In this case, the JFileChooser completely covers the Instruction window even when set using .testUI(). Not sure why the window manager doesn't honor the position set by PassFailJFrame, for now you can increase the number rows to increase the size of instruction window, so that the user running the manual test is aware that there are 2 windows.

File: test/jdk/javax/swing/JFileChooser/6698013/bug6698013.java [hunk 1]
```diff
@@ -45,13 +46,26 @@ public class bug6698013 extends JApplet {
...
-    }
-
-    public void init() {
-        JFileChooser chooser = new JFileChooser(new VirtualFileSystemView());
-        chooser.setCurrentDirectory(root);
-        chooser.showOpenDialog(null);
+        String instructions = """
+                1. Go into 'subdir' folder via double click
+                2. Return to parent directory
+                3. Go into 'subdir' folder: select 'subdir' folder and press the 'Open' button""";
```
Comment (TejeshR13) [inline test/jdk/javax/swing/JFileChooser/6698013/bug6698013.java:52]: Mentioning about Test Pass and Fail criteria would be beneficial, the instruction only says how to test, doesn't specify about Test PASS/FAIL criteria....?

**Review by TejeshR13 — COMMENTED** [review]
`rootFile`, `subdir `and `subdirFile` can be moved inside `VirtualFileSystemView` class ?

File: test/jdk/javax/swing/JFileChooser/6698013/bug6698013.java [hunk 2]
```diff
@@ -45,13 +46,26 @@ public class bug6698013 extends JApplet {
...
+
+        PassFailJFrame pfframe = PassFailJFrame.builder()
+                .title("bug6698013")
+                .instructions(instructions)
+                .rows(25)
```
Comment (DamonGuy) [inline test/jdk/javax/swing/JFileChooser/6698013/bug6698013.java:57]:
```suggestion
                .rows((int) INSTRUCTIONS.lines().count() + 2)
```

Your instructions frame has a lot of white space. Maybe worth incorporating this so you don't have to worry about adjusting the row count for each test.

**Review by DamonGuy — CHANGES_REQUESTED** [review]

**Review by honkar-jdk — APPROVED** [review]
LGTM.

Suggestion:
Since this test is problem-listed on macOS, mentioning it in PR description is helpful for reviewers as well as for future reference.

**Review by DamonGuy — APPROVED** [review]

**Review by TejeshR13 — APPROVED** [review]

**Pull request closed without merging**